// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: pkg/protocol/proto/connect/connect.ext.proto

package connectpb

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on Packet with the rules defined in the
// proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *Packet) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on Packet with the rules defined in the
// proto definition for this message. If any rules are violated, the result is
// a list of violation errors wrapped in PacketMultiError, or nil if none found.
func (m *Packet) ValidateAll() error {
	return m.validate(true)
}

func (m *Packet) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for Command

	// no validation rules for RequestId

	// no validation rules for Code

	// no validation rules for Message

	// no validation rules for Data

	if len(errors) > 0 {
		return PacketMultiError(errors)
	}

	return nil
}

// PacketMultiError is an error wrapping multiple validation errors returned by
// Packet.ValidateAll() if the designated constraints aren't met.
type PacketMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m PacketMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m PacketMultiError) AllErrors() []error { return m }

// PacketValidationError is the validation error returned by Packet.Validate if
// the designated constraints aren't met.
type PacketValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e PacketValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e PacketValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e PacketValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e PacketValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e PacketValidationError) ErrorName() string { return "PacketValidationError" }

// Error satisfies the builtin error interface
func (e PacketValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sPacket.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = PacketValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = PacketValidationError{}

// Validate checks the field values on SignInInput with the rules defined in
// the proto definition for this message. If any rules are violated, the first
// error encountered is returned, or nil if there are no violations.
func (m *SignInInput) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on SignInInput with the rules defined in
// the proto definition for this message. If any rules are violated, the
// result is a list of violation errors wrapped in SignInInputMultiError, or
// nil if none found.
func (m *SignInInput) ValidateAll() error {
	return m.validate(true)
}

func (m *SignInInput) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	// no validation rules for DeviceId

	// no validation rules for UserId

	// no validation rules for Token

	if len(errors) > 0 {
		return SignInInputMultiError(errors)
	}

	return nil
}

// SignInInputMultiError is an error wrapping multiple validation errors
// returned by SignInInput.ValidateAll() if the designated constraints aren't met.
type SignInInputMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m SignInInputMultiError) Error() string {
	msgs := make([]string, 0, len(m))
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m SignInInputMultiError) AllErrors() []error { return m }

// SignInInputValidationError is the validation error returned by
// SignInInput.Validate if the designated constraints aren't met.
type SignInInputValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e SignInInputValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e SignInInputValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e SignInInputValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e SignInInputValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e SignInInputValidationError) ErrorName() string { return "SignInInputValidationError" }

// Error satisfies the builtin error interface
func (e SignInInputValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sSignInInput.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = SignInInputValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = SignInInputValidationError{}
